runtime/src/main/java/org/corfudb/runtime/collections/ScopedTransaction.java

```diff
@@ -0,0 +1,157 @@
+package org.corfudb.runtime.collections;
+
+import com.google.protobuf.Message;
+import lombok.extern.slf4j.Slf4j;
+import org.corfudb.protocols.wireprotocol.Token;
+import org.corfudb.runtime.CorfuRuntime;
+
+import javax.annotation.Nonnull;
+import java.util.Arrays;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.Set;
+import java.util.function.BiFunction;
+import java.util.function.BiPredicate;
+import java.util.function.Function;
+import java.util.function.Predicate;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+@Slf4j
+public class ScopedTransaction implements
+        ReadTransactionApi, AutoCloseable {
+
+    private final ReadTransactionApi crud;
+    private final CorfuRuntime runtime;
+    private final String namespace;
+    private final Token txnSnapshot;
+    private final Map<
+            Table<? extends Message, ? extends Message, ? extends Message>,
+            Table> mapping;
+
+    ScopedTransaction(
+            @Nonnull final CorfuRuntime runtime,
+            @Nonnull final String namespace,
+            @Nonnull final IsolationLevel isolationLevel,
+            Table<?, ?, ?>... tables) {
+        this.crud = new ReadTransaction();
+        this.runtime = runtime;
+        this.namespace = namespace;
+        this.txnSnapshot = txBeginInternal(isolationLevel);
+        this.mapping = Arrays.stream(tables).collect(Collectors.toMap(
+                        table -> table,
+                        table -> table.generateImmutableView(txnSnapshot.getSequence())));
+    }
+
+    private Token txBeginInternal(IsolationLevel isolationLevel) {
+        log.trace("TxnContext: begin transaction in namespace {}", namespace);
+
+        if (isolationLevel.getTimestamp() != Token.UNINITIALIZED) {
+            return isolationLevel.getTimestamp();
+        }
+
+        return runtime
+                .getSequencerView()
+                .query()
+                .getToken();
+    }
+
+    @Nonnull
+    @Override
+    public <K extends Message, V extends Message, M extends Message> CorfuStoreEntry<K, V, M>
+    getRecord(@Nonnull Table<K, V, M> table, @Nonnull K key) {
+        return crud.getRecord(getTableSnapshot(table), key);
+    }
+
+    @Override
```

Comment:
For the rocksdb backed instance does this need to explicitly release the resources?